### PR TITLE
Release 4.3.0!

### PR DESCRIPTION
**Pull request by joeygrover**
Update to android studio, fixes, build flavors.

### CODE REVIEW

File: hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlReceiver.java [hunk 1]
```diff
@@ -6,24 +6,32 @@
...
 	public Class<? extends SdlRouterService> defineLocalSdlRouterClass() {
 		return com.sdl.hellosdlandroid.SdlRouterService.class;
 	}
 
-
+	@Override
+	public void onReceive(Context context, Intent intent) {
+		super.onReceive(context, intent); // Required if overriding this method
+
+		if(intent.getAction().equalsIgnoreCase(TransportConstants.START_ROUTER_SERVICE_ACTION)){
```
Comment (askirk) [inline hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlReceiver.java]: Got an null object for the action here when I changed the transport flavor to TCP, need to fix this, was thinking of replacing with
```
if(intent != null){
            if (intent.getBooleanExtra(RECONNECT_LANG_CHANGE, false)) {
                onSdlEnabled(context, intent);
            }
        }
```

Comment (brettywhite) [inline hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlReceiver.java]: The not null check should go around the line above the one you are referencing (the call to intent.getAction() above it:

```java
if(intent != null){
    if(intent.getAction().equalsIgnoreCase(TransportConstants.START_ROUTER_SERVICE_ACTION)){
        if (intent.getBooleanExtra(RECONNECT_LANG_CHANGE, false)) {
 				onSdlEnabled(context, intent);
 			}
                 }
      	}
```

Comment (brettywhite) [inline hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlReceiver.java]: https://github.com/smartdevicelink/hello_sdl_android/pull/17

Comment (askirk) [inline hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlReceiver.java]: @brettywhite the problem was `intent.getAction()` was null

Comment (brettywhite) [inline hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlReceiver.java]: @joeygrover the fix is in the branch [bugfix/nullobject](https://github.com/smartdevicelink/hello_sdl_android/tree/bugfix/nullObject) . If we can get that merged everything else looks good to go

**Pull request merged**